File: BI-TJV/semestral/rest-server/src/main/java/wrzecond/repository/TjvFoodRepository.java
package wrzecond.repository;

import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;
import wrzecond.entity.TjvAllergen;
import wrzecond.entity.TjvFood;

import java.util.List;

@Repository
public interface TjvFoodRepository extends TjvRepository<TjvFood> {

    @Query("SELECT allergen FROM TjvFood food JOIN food.allergens allergen WHERE food = :food")
    List<TjvAllergen> getAllergensInFood (TjvFood food);

}

File: BI-KOT/semestral/rest-server/src/main/kotlin/wrzecond/repository/TjvFoodRepository.kt
package wrzecond.repository

import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import wrzecond.entity.TjvAllergen
import wrzecond.entity.TjvFood

@Repository
interface TjvFoodRepository : TjvRepository<TjvFood> {

    @Query("SELECT allergen FROM TjvFood food JOIN food.allergens allergen WHERE food = :food")
    fun getAllergensInFood (food: TjvFood) : List<TjvAllergen>

}
